src/NetscriptFunctions.ts

```diff
@@ -1647,22 +1646,26 @@ export const ns: InternalAPI<NSFull> = {
       return Promise.resolve(false);
     }
     return new Promise(function (resolve) {
-      $.get(
-        url,
-        function (data) {
-          const res = server.writeToContentFile(target, data);
-          if (res.overwritten) {
+      fetch(url)
```

Comment:
IMO the flow of all of this would be simpler if you promoted wget itself (inside the ctx wrapper) to async. Then you can just return true/false directly, and catch the error on fetch to do error handling instead of it being buried later.